core/app/finders/spree/option_values/find_filterable.rb

```diff
@@ -0,0 +1,24 @@
+module Spree
+  module OptionValues
+    class FindFilterable
+      def initialize(scope: OptionValue.spree_base_scopes, products_scope: Product.spree_base_scopes)
+        @scope = scope
+        @products_scope = products_scope
+      end
+
+      def execute
+        scope.
+          filterable.
+          where(variants: { product_id: products_ids })
```

Comment:
@mad-eel can we avoid passing IDs and maybe use `.merge` ?

Reply:
@damianlegawiec In basic usage `merge` seems to work. But when giving some more complex scope, it fails. As an example: `in_taxon` includes `classifications` which is not present on the option values scope causing an error.

We can rewrite it to be less generic and accept `taxon` and `currency`, but I'm still running into issues with the `select distinct` list not matching the `order by` list.

I'm not really sure how to go from here.
cc @MatthewKennedy @Vegann